examples/bindings/quickstart_sparse_string/Program.cs

```diff
@@ -0,0 +1,330 @@
+            Methods.tiledb_dimension_free(d2);
+            Methods.tiledb_attribute_free(&a1);
+            Methods.tiledb_domain_free(&domain);
+            Methods.tiledb_array_schema_free(&array_schema);
+        }
+
+        static unsafe void WriteArray(tiledb_ctx_t* ctx, tiledb_array_t* array)
+        {
+            // Open array for writing
+            Methods.tiledb_array_open(ctx, array, tiledb_query_type_t.TILEDB_WRITE);
+            CheckError(ctx);
+
+            // Allocate data to write into array
+            byte[] data_rows = Encoding.ASCII.GetBytes("abbc");
+            UInt64 data_rows_size = 4;
+
+            UInt64[] data_rows_offsets = { 0, 1, 3 };
+            UInt64 data_rows_offsets_size = (UInt64)data_rows_offsets.Length * sizeof(UInt64);
+
+            Int32[] data_cols_coords = { 1, 4, 3 };
+            UInt64 data_cols_coords_size = (UInt64)data_cols_coords.Length * sizeof(Int32);
+
+            Int32[] data = { 5, 6, 7 };
+            UInt32 data_size = (UInt32)data.Length * sizeof(Int32);
+
+            var attr_name = (sbyte*)Marshal.StringToHGlobalAnsi("a1");
+            var rows_name = (sbyte*)Marshal.StringToHGlobalAnsi("rows");
+            var cols_name = (sbyte*)Marshal.StringToHGlobalAnsi("cols");
+            
+            // Create write query
+            tiledb_query_t* query;
+            Methods.tiledb_query_alloc(ctx, array, tiledb_query_type_t.TILEDB_WRITE, &query);
+            CheckError(ctx);
+            
+            Methods.tiledb_query_set_layout(ctx, query, tiledb_layout_t.TILEDB_UNORDERED);
+            CheckError(ctx);
+
+            var data_handle = GCHandle.Alloc(data, GCHandleType.Pinned);
+            var data_size_handle = GCHandle.Alloc(data_size, GCHandleType.Pinned);
+            Methods.tiledb_query_set_data_buffer(ctx, query, attr_name, 
+                data_handle.AddrOfPinnedObject().ToPointer(), (ulong*)data_size_handle.AddrOfPinnedObject()
+            );
+            data_handle.Free();
+            data_size_handle.Free();
+            CheckError(ctx);
+
+            data_handle = GCHandle.Alloc(data_rows, GCHandleType.Pinned);
+            data_size_handle = GCHandle.Alloc(data_rows_size, GCHandleType.Pinned);
+            Methods.tiledb_query_set_data_buffer(ctx, query, rows_name,
+                data_handle.AddrOfPinnedObject().ToPointer(), (ulong*)data_size_handle.AddrOfPinnedObject()
+            );
+            data_handle.Free();
+            data_size_handle.Free();
+            CheckError(ctx);
+
+            data_handle = GCHandle.Alloc(data_rows_offsets, GCHandleType.Pinned);
+            data_size_handle = GCHandle.Alloc(data_rows_offsets_size, GCHandleType.Pinned);
+            Methods.tiledb_query_set_offsets_buffer(ctx, query, rows_name,
+                (ulong*)data_handle.AddrOfPinnedObject(), (ulong*)data_size_handle.AddrOfPinnedObject()
+            );
+            data_handle.Free();
+            data_size_handle.Free();

```

Comment:
Unless I am missing something about how GCHandle works, I believe these handles need to be kept alive until after the `tiledb_query_submit` call -- the size pointers are held internally until after query_submit returns.